### ISSUE TITLE
Windows Powershell (x64) Session will not start - seems like #118289

### ISSUE THREAD

**Issue by ro-epp** [PowerShell/vscode-powershell#4012]
### Prerequisites

- [X] I have written a descriptive issue title.
- [X] I have searched all [issues](https://github.com/PowerShell/vscode-powershell/issues?q=is%3Aissue) to ensure it has not already been reported.
- [X] I have read the [troubleshooting](https://github.com/PowerShell/vscode-powershell/blob/master/docs/troubleshooting.md) guide.
- [X] I am sure this issue is with the _extension itself_ and does not reproduce in a standalone [PowerShell](https://github.com/PowerShell/PowerShell/issues/new/choose) instance.
- [X] I have verified that I am using the latest version of Visual Studio Code and the PowerShell extension.
- [ ] If this is a security issue, I have read the [security issue reporting guidance](https://github.com/PowerShell/vscode-powershell/blob/master/SECURITY.md).

### Summary

My issue is that the Windows PowerShell Session (x64) will not finish loading/launching. 32bit will load, but does not have access to 64bit tools. I have installed the last December version and the Windows PowerShell Session (x64) successfully loaded, 2022.5.0 did not load the powershell session. I even changed "set ELECTRON_RUN_AS_NODE=1" to 0,
Below are the logs for the extension service. 

### PowerShell Version

```console
Name                           Value                                                                              
----                           -----                                                                              
PSVersion                      5.1.19041.1682                                                                     
PSEdition                      Desktop                                                                            
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}                                                            
BuildVersion                   10.0.19041.1682                                                                    
CLRVersion                     4.0.30319.42000                                                                    
WSManStackVersion              3.0                                                                                
PSRemotingProtocolVersion      2.3                                                                                
SerializationVersion           1.1.0.1
```

### Visual Studio Code Version

```console
1.67.2
c3511e6c69bb39013c4a4b7b9566ec1ca73fc4d5
x64
```

### Extension Version

```console
ms-vscode.powershell@2022.5.1
```

### Steps to Reproduce

open VS Code and open/create a ps1 file to get the extension to load.

### Logs

6/7/2022 2:42:20 PM [NORMAL] - Language server starting --
6/7/2022 2:42:20 PM [NORMAL] -     PowerShell executable: C:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe
6/7/2022 2:42:20 PM [NORMAL] - 
WARNING: In development mode but PowerShellEditorServices dev module path cannot be found (or has not been built yet): c:\Users\re1\.vscode\extensions\PowerShellEditorServices\module

6/7/2022 2:42:20 PM [NORMAL] -     PowerShell args: -NoProfile -ExecutionPolicy Bypass -Command Import-Module 'c:\Users\re1\.vscode\extensions\ms-vscode.powershell-2022.5.1\modules\PowerShellEditorServices\PowerShellEditorServices.psd1'; Start-EditorServices -HostName 'Visual Studio Code Host' -HostProfileId 'Microsoft.VSCode' -HostVersion '2022.5.1' -AdditionalModules @('PowerShellEditorServices.VSCode') -BundledModulesPath 'c:\Users\re1\.vscode\extensions\ms-vscode.powershell-2022.5.1\modules' -EnableConsoleRepl -StartupBanner '=====> PowerShell Integrated Console v2022.5.1 <=====
' -LogLevel 'Normal' -LogPath 'c:\Users\re1\.vscode\extensions\ms-vscode.powershell-2022.5.1\logs\1654630940-2a4fc57d-2bd6-4f0f-8f6d-a94ae1f80f271654630932896\EditorServices.log' -SessionDetailsPath 'c:\Users\re1\.vscode\extensions\ms-vscode.powershell-2022.5.1\sessions\PSES-VSCode-20164-991690' -FeatureFlags @() 
6/7/2022 2:42:20 PM [NORMAL] - Waiting for session file
6/7/2022 2:42:20 PM [NORMAL] - powershell.exe started.
6/7/2022 2:42:20 PM [NORMAL] -     PowerShell Editor Services args: Import-Module 'c:\Users\re1\.vscode\extensions\ms-vscode.powershell-2022.5.1\modules\PowerShellEditorServices\PowerShellEditorServices.psd1'; Start-EditorServices -HostName 'Visual Studio Code Host' -HostProfileId 'Microsoft.VSCode' -HostVersion '2022.5.1' -AdditionalModules @('PowerShellEditorServices.VSCode') -BundledModulesPath 'c:\Users\re1\.vscode\extensions\ms-vscode.powershell-2022.5.1\modules' -EnableConsoleRepl -StartupBanner '=====> PowerShell Integrated Console v2022.5.1 <=====
' -LogLevel 'Normal' -LogPath 'c:\Users\re1\.vscode\extensions\ms-vscode.powershell-2022.5.1\logs\1654630940-2a4fc57d-2bd6-4f0f-8f6d-a94ae1f80f271654630932896\EditorServices.log' -SessionDetailsPath 'c:\Users\re1\.vscode\extensions\ms-vscode.powershell-2022.5.1\sessions\PSES-VSCode-20164-991690' -FeatureFlags @() 
6/7/2022 2:42:22 PM [NORMAL] - Session file found
6/7/2022 2:42:22 PM [NORMAL] - Registering terminal close callback
6/7/2022 2:42:22 PM [NORMAL] - Registering terminal PID log callback
6/7/2022 2:42:22 PM [NORMAL] - Language server started.
6/7/2022 2:42:22 PM [NORMAL] - powershell.exe PID: 9100
6/7/2022 2:42:22 PM [NORMAL] - Connecting to language service on pipe \\.\pipe\PSES_hltubtat.brd...
6/7/2022 2:42:22 PM [NORMAL] - {"status":"started","languageServiceTransport":"NamedPipe","languageServicePipeName":"\\\\.\\pipe\\PSES_hltubtat.brd","debugServiceTransport":"NamedPipe","debugServicePipeName":"\\\\.\\pipe\\PSES_3mf5vomg.hfa"}
6/7/2022 2:42:22 PM [NORMAL] - Language service connected.
6/7/2022 2:47:23 PM [WARNING] - Invalid Version: null

**Comment by SydneyhSmith** [comment]
Thanks @ro-epp does this reproduce with PowerShell 7.x or just Windows PowerShell? Also have you tried uninstalling and reinstalling VSCode...seems like something strange may be going on with VSCode...thanks

**Comment by ShikeChen-MS** [comment]
Same here, I am on PowerShell 7.2.4 and the integrated console does not start.

**Comment by mwander-chwy** [comment]
Same here, I am on PowerShell 7.2.4 and the integrated console does not start.
![Capture](https://user-images.githubusercontent.com/92752646/173706910-5d8ddd06-fea0-4ab2-8798-e965e78cb35d.PNG)

After some time I press Ctrl-C and get this:
![Capture2](https://user-images.githubusercontent.com/92752646/173707088-55f66c8d-062a-4484-86e3-b81c1d587fb3.PNG)
![Capture3](https://user-images.githubusercontent.com/92752646/173707101-66649ba8-b5c9-4ec3-a477-241750a278c5.PNG)

My log file looks practically the same as the above log file.  Except I'm using 7.2.4

**Comment by ShikeChen-MS** [comment]
Just tried preview, same result:
![1](https://user-images.githubusercontent.com/100958637/173906493-615a7deb-30e2-47a2-8cd5-d6cfe55ff03f.jpg)

**Comment by andschwa** [comment]
I see `OnCancelKeyPress` in your log...did you hit Ctrl-C perhaps?

**Comment by ro-epp** [comment]
> 

I have not tried 7.2.4 because nothing else in my environment uses it and I cannot clearly verify what will not work on 5.1 (I am probably just misunderstanding the notes).

**Comment by ro-epp** [comment]
Update:
There was a new version of VSCode that was available, 1.68.0. That seems to have cleared up the issue with the x64 session starting. Hopefully 1.68.1 does not make the issue recur.

**Comment by ShikeChen-MS** [comment]
> I see `OnCancelKeyPress` in your log...did you hit Ctrl-C perhaps?

Yes, I have to hit ctrl-c otherwise, it will just hanging there with no output

**Comment by ShikeChen-MS** [comment]
> Update: There was a new version of VSCode that was available, 1.68.0. That seems to have cleared up the issue with the x64 session starting. Hopefully 1.68.1 does not make the issue recur.

I am currently on VSCode 1.68.1.
As following info shows (copied from vscode)
```
Version: 1.68.1 (user setup)
Commit: 30d9c6cd9483b2cc586687151bcbcd635f373630
Date: 2022-06-14T12:48:58.283Z
Electron: 17.4.7
Chromium: 98.0.4758.141
Node.js: 16.13.0
V8: 9.8.177.13-electron.0
OS: Windows_NT x64 10.0.22000
```

**Comment by opmdsm** [comment]
I also have the same issue. Windows PowerShell (x86) loads, Windows PowerShell (x64) doesn't. I'm on latest Visual Studio Code 1.68.1. Uninstall PowerShell extension and install it again doesn't help. Uninstall Visual Studio Code and install it again also doesn't help. I remember that at very first install of PowerShell extension, Visual Studio Code asked me if I agree with something (with y/n option on console/terminal) and this doesn't happen if I uninstall and install again PowerShell extension or Visual Studio Code. So to me looks like some info is kept at OS level that prevents PowerShell (x64) to load.

![VSC](https://user-images.githubusercontent.com/107676477/174263036-c54953c9-057d-4dbc-a564-d658fc31f2ea.PNG)

**Comment by mrjcap** [comment]
I had the same issue (powershell 7.2.4) for several days and I decided to dig a little deeper.

In my case, when I was starting vscode, pwsh.exe was creating a child process of powershell.exe
	
After hours of troubleshooting, reinstalling, and reconfiguring pretty much everything. I still couldn't figure out what causing the powershell.exe process to launch.

Finally, I removed the following path from the environment variables: C:\Windows\System32\WindowsPowerShell\v1.0\
And: %SystemRoot%\system32\WindowsPowerShell\v1.0\Modules from PSModulePath.

Then I was able to start integrated console successfully.

Another thing I noticed when I did a trace with procmon. There was a lot of activity (37000 read operations) with the StartupProfileData-Interactive file.

**Comment by opmdsm** [comment]
> I had the same issue (powershell 7.2.4) for several days and I decided to dig a little deeper.
> 
> In my case, when I was starting vscode, pwsh.exe was creating a child process of powershell.exe
> 
> After hours of troubleshooting, reinstalling, and reconfiguring pretty much everything. I still couldn't figure out what causing the powershell.exe process to launch.
> 
> Finally, I removed the following path from the environment variables: C:\Windows\System32\WindowsPowerShell\v1.0 And: %SystemRoot%\system32\WindowsPowerShell\v1.0\Modules from PSModulePath.
> 
> Then I was able to start integrated console successfully.
> 
> Another thing I noticed when I did a trace with procmon. There was a lot of activity (37000 read operations) with the StartupProfileData-Interactive file.

Thanks for the tip. I don't have C:\Windows\System32\WindowsPowerShell\v1.0 in PSModulePath but I have the other one: %SystemRoot%\system32\WindowsPowerShell\v1.0\Modules. Removing this causes both of them to not start (PowerShell x86 & x64). Put it back, x86 working again, x64 still no go.

**Comment by mwander-chwy** [comment]
It worked for me so thank you @mrjcap worked flawlessly for me !!!!!! Finally back up and running!

**Comment by mrjcap** [comment]
> > I had the same issue (powershell 7.2.4) for several days and I decided to dig a little deeper.
> > In my case, when I was starting vscode, pwsh.exe was creating a child process of powershell.exe
> > After hours of troubleshooting, reinstalling, and reconfiguring pretty much everything. I still couldn't figure out what causing the powershell.exe process to launch.
> > Finally, I removed the following path from the environment variables: C:\Windows\System32\WindowsPowerShell\v1.0 And: %SystemRoot%\system32\WindowsPowerShell\v1.0\Modules from PSModulePath.
> > Then I was able to start integrated console successfully.
> > Another thing I noticed when I did a trace with procmon. There was a lot of activity (37000 read operations) with the StartupProfileData-Interactive file.
> 
> Thanks for the tip. I don't have C:\Windows\System32\WindowsPowerShell\v1.0 in PSModulePath but I have the other one: %SystemRoot%\system32\WindowsPowerShell\v1.0\Modules. Removing this causes both of them to not start (PowerShell x86 & x64). Put it back, x86 working again, x64 still no go.

Sorry about that. 

I should've mentioned that  C:\Windows\System32\WindowsPowerShell\v1.0 is in PATH. Not in PSModulePath. 

I have updated my reply.

**Comment by andschwa** [comment]
What do you all have configured for your PowerShell additional exe paths?

```
"powershell.powerShellAdditionalExePaths": {}
```

and your PowerShell default version?

```
"powershell.powerShellDefaultVersion": ""
```

Thanks!

**Comment by mrjcap** [comment]
![Capture](https://user-images.githubusercontent.com/3140504/174331692-79ea47f8-c152-4e07-9e9c-db81fc2f6ff8.PNG)

**Comment by ShikeChen-MS** [comment]
> I had the same issue (powershell 7.2.4) for several days and I decided to dig a little deeper.
> 
> In my case, when I was starting vscode, pwsh.exe was creating a child process of powershell.exe
> 
> After hours of troubleshooting, reinstalling, and reconfiguring pretty much everything. I still couldn't figure out what causing the powershell.exe process to launch.
> 
> Finally, I removed the following path from the environment variables: C:\Windows\System32\WindowsPowerShell\v1.0\ from PATH. And: %SystemRoot%\system32\WindowsPowerShell\v1.0\Modules from PSModulePath.
> 
> Then I was able to start the integrated console successfully.
> 
> Another thing I noticed when I did a trace with procmon. There was a lot of activity (37000 read operations) with the StartupProfileData-Interactive file.

Thanks, this workaround works for me.

However, with that said, I am still wondering what's the root cause is and hope it can be solved. Removing staff from environment variable doesn't seems to be a good permanant solution.

**Comment by andschwa** [comment]
Hopefully with this information we may be able to reproduce the problem locally, which is 90% of fixing it.

**Comment by mrjcap** [comment]
Another temporary solution is to downgrade the extension. I tried every version from 2022.1.0 to 2022.5.1.

All the 2022.xx.xx versions had this issue. The only working version I was able to find is: 2021.12.0.

I wouldn't recommend it though.

**Comment by opmdsm** [comment]
> > > I had the same issue (powershell 7.2.4) for several days and I decided to dig a little deeper.
> > > In my case, when I was starting vscode, pwsh.exe was creating a child process of powershell.exe
> > > After hours of troubleshooting, reinstalling, and reconfiguring pretty much everything. I still couldn't figure out what causing the powershell.exe process to launch.
> > > Finally, I removed the following path from the environment variables: C:\Windows\System32\WindowsPowerShell\v1.0 And: %SystemRoot%\system32\WindowsPowerShell\v1.0\Modules from PSModulePath.
> > > Then I was able to start integrated console successfully.
> > > Another thing I noticed when I did a trace with procmon. There was a lot of activity (37000 read operations) with the StartupProfileData-Interactive file.
> > 
> > 
> > Thanks for the tip. I don't have C:\Windows\System32\WindowsPowerShell\v1.0 in PSModulePath but I have the other one: %SystemRoot%\system32\WindowsPowerShell\v1.0\Modules. Removing this causes both of them to not start (PowerShell x86 & x64). Put it back, x86 working again, x64 still no go.
> 
> Sorry about that.
> 
> I should've mentioned that C:\Windows\System32\WindowsPowerShell\v1.0 is in PATH. Not in PSModulePath.
> 
> I have updated my reply.

Thanks, but still doesn't work for me. Downgrading extension to 2021.12.0 make it works again.

**Comment by andschwa** [comment]
Ok, got a lot going on here, trying to summarize it:

* The original issue posted by @ro-epp was that Windows PowerShell x64 was not starting, this was [resolved](https://github.com/PowerShell/vscode-powershell/issues/4012#issuecomment-1156840934) for them with a VS Code update to 1.68.0
* @ShikeChen-MS, @mrjcap, and @mwander-chwy all have a separate but similar issue where PowerShell Core 7.2.4 is not loading, they [resolved](https://github.com/PowerShell/vscode-powershell/issues/4012#issuecomment-1159062875) their issues by realizing that `pwsh.exe` was trying to start `powershell.exe`, and failing, and so by removing the path to `powershell.exe` from their system `PATH` environment. 
  * @mrjcap thank you for sharing your configuration. You have specifically set your extension to try to load Windows PowerShell by setting `powershell.powerShellDefaultVersion` to `PowerShell (×64)`. Instead of editing `PATH` (and `PSModulePath`) can you please reset your extension configuration to load PowerShell Core AKA `pwsh.exe`? _I do not believe you need to edit your environment variables nor downgrade the extension, but you do need to correct your extension configuration._
  * @ShikeChen-MS and @mwander-chwy can you both check the same setting please, since your issue appears to be the same as @mrjcap?
* @opmdsm you seem to have a completely separate issue from any of the others. You're trying to get `PowerShell (×64)` (Windows PowerShell) to load, but indicate you were asked something at startup. My _guess_ was you were asked if you trusted the editor services startup script, and presumably said no. This will lead to it failing to load silently. Can you please try following [these instructions](https://github.com/PowerShell/vscode-powershell/issues/3729#issuecomment-1000395746) to trust the script?

I have a feeling that the unexpected behavior where Windows PowerShell is being started instead of PowerShell Core can be traced to the accidentally breaking change made to `powershell.powerShellAdditionalExePaths` in #3919. Please see #3978 for more thorough information on setting up your available PowerShell sessions.

**Comment by andschwa** [comment]
What are test cases 2 and 3 attempting to test?

**Comment by mrjcap** [comment]
The non-working state.

**Comment by andschwa** [comment]
What if you set up test cases 2 and 3 to explicitly load PowerShell Core?

**Comment by mrjcap** [comment]
I will have to test it tomorrow and let you know.

**Comment by opmdsm** [comment]
> Ok, got a lot going on here, trying to summarize it:
> 
> * The original issue posted by @ro-epp was that Windows PowerShell x64 was not starting, this was [resolved](https://github.com/PowerShell/vscode-powershell/issues/4012#issuecomment-1156840934) for them with a VS Code update to 1.68.0
> * @ShikeChen-MS, @mrjcap, and @mwander-chwy all have a separate but similar issue where PowerShell Core 7.2.4 is not loading, they [resolved](https://github.com/PowerShell/vscode-powershell/issues/4012#issuecomment-1159062875) their issues by realizing that `pwsh.exe` was trying to start `powershell.exe`, and failing, and so by removing the path to `powershell.exe` from their system `PATH` environment.
>   
>   * @mrjcap thank you for sharing your configuration. You have specifically set your extension to try to load Windows PowerShell by setting `powershell.powerShellDefaultVersion` to `PowerShell (×64)`. Instead of editing `PATH` (and `PSModulePath`) can you please reset your extension configuration to load PowerShell Core AKA `pwsh.exe`? _I do not believe you need to edit your environment variables nor downgrade the extension, but you do need to correct your extension configuration._
>   * @ShikeChen-MS and @mwander-chwy can you both check the same setting please, since your issue appears to be the same as @mrjcap?
> * @opmdsm you seem to have a completely separate issue from any of the others. You're trying to get `PowerShell (×64)` (Windows PowerShell) to load, but indicate you were asked something at startup. My _guess_ was you were asked if you trusted the editor services startup script, and presumably said no. This will lead to it failing to load silently. Can you please try following [these instructions](https://github.com/PowerShell/vscode-powershell/issues/3729#issuecomment-1000395746) to trust the script?
> 
> I have a feeling that the unexpected behavior where Windows PowerShell is being started instead of PowerShell Core can be traced to the accidentally breaking change made to `powershell.powerShellAdditionalExePaths` in #3919. Please see #3978 for more thorough information on setting up your available PowerShell sessions.

Yes, I have been asked but only once, at the very first extension installation. Don't recall exactly what was the question. After that, no matter what I did (uninstall extension or VS Code and install again) I didn't get it anymore. The script is trusted, tried your suggestions with same results, PowerShell x64 doesn't load. I've checked powershell.powerShellDefaultVersion and is empty. Should I have something in it?

**Comment by mrjcap** [comment]
@andschwa,

I changed the powerShellDefaultVersion to PowerShell Core and now it seems that I cannot reproduce the issue.

**Comment by opmdsm** [comment]
I did some more tests on 2 different machines: Win 10 and Win 2022. The question that I was talking about in previous posts is not when installing PowerShell extension (my bad), but when opening an existing ps1 file or PowerShell workspace. In both cases I answer yes, I trust them.

The steps were exactly the same as follows:
1. Install VS Code (Setup) and start as admin
2. Install PowerShell extension
3. Open an existing workspace or ps1 file. I tried both and trust them.

The difference:
On Win 2022, everything went ok.
On Win 10, I got what is an attachment. In addition to that, PowerShell extension is not loading.

![Capture](https://user-images.githubusercontent.com/107676477/174756085-e2a822a8-01a6-4846-b3e2-123e37b28765.PNG)

**Comment by andschwa** [comment]
> I did some more tests on 2 different machines: Win 10 and Win 2022. The question that I was talking about in previous posts is not when installing PowerShell extension (my bad), but when opening an existing ps1 file or PowerShell workspace. In both cases I answer yes, I trust them.
> 
> The steps were exactly the same as follows:
> 
> 1. Install VS Code (Setup) and start as admin
> 2. Install PowerShell extension
> 3. Open an existing workspace or ps1 file. I tried both and trust them.
> 
> The difference: On Win 2022, everything went ok. On Win 10, I got what is in below picture. In addition to that, PowerShell extension is not loading.
> 
> ![Capture](https://github.com/assets/107676477/e2a822a8-01a6-4846-b3e2-123e37b28765)
> 
> After some time I got this:
> 
> ![Capture1](https://github.com/assets/107676477/7502f95d-2f90-4945-bca4-be766e58ab2c)
> 
> And here are the logs (I replaced my user name with <USER_NAME>)
> 
> [Logs.txt](https://github.com/PowerShell/vscode-powershell/files/8947317/Logs.txt)
> 
> Maybe it helps.
> 
> After waiting additional time, finally PowerShell session was started. This is as close I can get to reproduce the issue. On my PC, the loading is still not happening.

On your Windows 10 machine, can you try installing WMF 5.1 per [here](https://github.com/PowerShell/vscode-powershell/blob/master/docs/troubleshooting.md#using-windows-powershell-51)?

**Comment by mrjcap** [comment]
> > @andschwa,
> > I changed the powerShellDefaultVersion to PowerShell Core and now it seems that I cannot reproduce the issue.
> 
> @mrjcap are you opening Code via the command-line? As in, are you in a `pwsh` session, and running `code`, so the process tree looks like `pwsh->code->pwsh` (and when it was "broken" it looked like `pwsh->code->powershell`)? If so I think the reason it wasn't working was due to #3886, and I bet you could get "PowerShell (x64)" i.e. `powershell` to work if the tree was instead `powershell->code->powershell` (or take out the first PowerShell process completely and start Code from its shortcut).

No I am not. I am starting it from the shortcut.

Here is a screenshot from procmon:

![not_working](https://user-images.githubusercontent.com/3140504/174958569-fa641bd5-95be-4ecc-a949-659b5ae83d67.PNG)

And here as screenshot when it's working:


![working](https://user-images.githubusercontent.com/3140504/174958685-643651cf-c707-447c-9164-cb5e36b368c8.PNG)


And a comparison of the read/write operations:

![comparison](https://user-images.githubusercontent.com/3140504/174958842-d5a75713-c5c9-4601-ad24-53d563a36878.PNG)

As you can see it only take 3 read operations when it's working and around 30000 operations when it's not.

I hope this helps.

**Comment by opmdsm** [comment]
> > I did some more tests on 2 different machines: Win 10 and Win 2022. The question that I was talking about in previous posts is not when installing PowerShell extension (my bad), but when opening an existing ps1 file or PowerShell workspace. In both cases I answer yes, I trust them.
> > The steps were exactly the same as follows:
> > 
> > 1. Install VS Code (Setup) and start as admin
> > 2. Install PowerShell extension
> > 3. Open an existing workspace or ps1 file. I tried both and trust them.
> > 
> > The difference: On Win 2022, everything went ok. On Win 10, I got what is in below picture. In addition to that, PowerShell extension is not loading.
> > ![Capture](https://github.com/assets/107676477/e2a822a8-01a6-4846-b3e2-123e37b28765)
> > After some time I got this:
> > ![Capture1](https://github.com/assets/107676477/7502f95d-2f90-4945-bca4-be766e58ab2c)
> > And here are the logs (I replaced my user name with <USER_NAME>)
> > [Logs.txt](https://github.com/PowerShell/vscode-powershell/files/8947317/Logs.txt)
> > Maybe it helps.
> > After waiting additional time, finally PowerShell session was started. This is as close I can get to reproduce the issue. On my PC, the loading is still not happening.
> 
> @opmdsm On your Windows 10 machine, can you try installing WMF 5.1 per [here](https://github.com/PowerShell/vscode-powershell/blob/master/docs/troubleshooting.md#using-windows-powershell-51)?

Hmm... Which one? They say on Win 10 is built in...

![image](https://user-images.githubusercontent.com/107676477/174972139-edb80248-fba7-4e0b-af6a-b5f70123eaaf.png)

**Comment by andschwa** [comment]
@opmdsm you are right, WMF 5.1 looks to be built-in for your Windows version now. And indeed, that looks like a permissions issue. It is strange, I see you answered `A` so it should have run.

**Comment by andschwa** [comment]
@mrjcap You are able to repro this on a VM, correct? If so, can you get a repro on the VM with multiple cores enabled, and then change its configuration to be single core (and reboot it) and see if it still repros? That `StartupProfileData-NonInteractive` comes from the .NET JIT optimizer, which is only enabled on multi-core machines, so I am hoping this test will reveal if it is indeed a problem with that, or if it's a red herring.

**Comment by mrjcap** [comment]
@andschwa,

No, I cannot reproduce it on a VM. The non-working version is on a physical machine.

**Comment by andschwa** [comment]
Well dang 😓

**Comment by andschwa** [comment]
@mrjcap How about try deleting `~\AppData\Local\Microsoft\PowerShellStartupProfileData-Nonlnteractive` (the path it's writing to 30,000+ times). We think the documentation says that if it doesn't exist, it won't use it.

**Comment by mrjcap** [comment]
Yeah, I have done that. When I re-run VS Code, the file is recreated.

**Comment by opmdsm** [comment]
> @opmdsm you are right, WMF 5.1 looks to be built-in for your Windows version now. And indeed, that looks like a permissions issue. It is strange, I see you answered `A` so it should have run.

FYI I have installed PowerShell 7.2.5 (x64) and it works. Still built in Windows PowerShell 5.1 (x64) does not load.

**Comment by mavaddat** [comment]
I am getting the same error as long as I am using Windows 10.
![ezgif com-optimize](https://github.com/PowerShell/vscode-powershell/assets/5055400/31080ef5-ef4d-49ee-9bd6-ee2e5def139f)

PSVersionTable
```pwsh
Name                           Value                                                                                   
----                           -----                                                                                   
PSVersion                      5.1.19041.2673                                                                          
PSEdition                      Desktop                                                                                 
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}                                                                 
BuildVersion                   10.0.19041.2673                                                                         
CLRVersion                     4.0.30319.42000                                                                         
WSManStackVersion              3.0                                                                                     
PSRemotingProtocolVersion      2.3                                                                                     
SerializationVersion           1.1.0.1
```
VS Code Version
```
1.80.0-insider
x64
```

Extension verison
```
ms-vscode.powershell@2023.6.0
```

| System Info | Value |
|---|---|
OS Name | Microsoft Windows 10 Enterprise
OS Version | 10.0.19044 N/A Build 19044
OS Manufacturer | Microsoft Corporation
OS Configuration | Member Workstation
OS Build Type | Multiprocessor Free
Product ID | 00329-00000-00003-AA531
Original Install Date | 2023-04-12, 5
System Boot Time | 2023-06-16, 1
System Model | Latitude 5430
System Type | x64-based PC
Processor(s) | 1 Processor(s) Installed.
[01] | Intel64 Family 6 Model 154 Stepping 4 GenuineIntel ~1600 Mhz

**Comment by marius-oloeru** [comment]
I have fixed this by chance. OneDrive corrupted somehow a file from Documents folder: C:\Users\<YOUR_USER_NAME>\OneDrive \Documents\WindowsPowerShell\Modules\PackageManagement\1.4.7. Once I ran command chkdsk on C drive, everything was fine. Hope it helps.

**Comment by ZERO-S1GNAL** [comment]
Just dropping by to say it's still an issue and the solution works. VSC v1.98.1, PS extension v2025.0.0 - I also had Security module problems (same as mentioned in one of the posts related to this issue), which are solved by loading it with -Force param.
